Flaky Regressions due to reach the max number of threads with FullTeaching test suite

A couple of days ago, we encountered a failure during the update process. The failure occurred due to an issue instantiating a new thread in Maven Surefire. Before starting the debugging process, I noticed a significant number of zombie processes in our virtual machine:

![Zombie Processes](https://github.com/user-attachments/assets/62a4492e-661e-427f-8aca-b5806fc5dda9)

These processes originated from the JDK of the Jenkins slave:

![JDK Jenkins Slave](https://github.com/user-attachments/assets/47ac0661-0700-4218-b000-737fde071b0c)

#### Jenkins Trace Log
The test execution was unable to create the log file, but the content of the Jenkins trace is as follows:

```log
+ /home/jenkins/workspace/retorch-st-fullteaching_main/retorchfiles/scripts/tjoblifecycles/tjob-testexecution.sh tjobn 2 https://full-teaching-/ 5000 'FullTeachingEndToEndRESTTests#sessionRestOperations'
Using the provided URL
The URL IS https://full-teaching-/
The HOST_IP is: 172.22.0.1 and PORT 5000 for the TJOB tjobn
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------< com.fullteaching:e2e.no-elastest >------------------
[INFO] Building com.fullteaching:e2e.no-elastest 1.1.1-SNAPSHOT
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- resources:3.3.1:resources (default-resources) @ e2e.no-elastest ---
[INFO] skip non existing resourceDirectory /home/jenkins/workspace/retorch-st-fullteaching_main/src/main/resources
[INFO] 
[INFO] --- compiler:3.13.0:compile (default-compile) @ e2e.no-elastest ---
[INFO] Nothing to compile - all classes are up to date.
[INFO] 
[INFO] --- resources:3.3.1:testResources (default-testResources) @ e2e.no-elastest ---
[INFO] Copying 6 resources from src/test/resources to target/test-classes
[INFO] 
[INFO] --- compiler:3.13.0:testCompile (default-testCompile) @ e2e.no-elastest ---
[INFO] Nothing to compile - all classes are up to date.
[INFO] 
[INFO] --- jacoco:0.8.12:prepare-agent (jacoco-initialize) @ e2e.no-elastest ---
[INFO] argLine set to -javaagent:/root/.m2/repository/org/jacoco/org.jacoco.agent/0.8.12/org.jacoco.agent-0.8.12-runtime.jar=destfile=/home/jenkins/workspace/retorch-st-fullteaching_main/target/coverage-reports/jacoco-unit.exec,append=true,excludes=*MethodAccess,output=file
[INFO] 
[INFO] --- surefire:3.3.1:test (default-test) @ e2e.no-elastest ---
[INFO] Using auto detected provider org.apache.maven.surefire.junitplatform.JUnitPlatformProvider
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[ERROR] Unable to create test class 'com.fullteaching.e2e.no_elastest.functional.test.media.FullTeachingEndToEndRESTTests'
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  4.858 s
[INFO] Finished at: 2024-07-29T11:54:46+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.3.1:test (default-test) on project e2e.no-elastest: 
[ERROR] 
[ERROR] Please refer to /home/jenkins/workspace/retorch-st-fullteaching_main/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] There was an error in the forked process
[ERROR] Unable to create test class 'com.fullteaching.e2e.no_elastest.functional.test.media.FullTeachingEndToEndRESTTests'
[ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: There was an error in the forked process
[ERROR] Unable to create test class 'com.fullteaching.e2e.no_elastest.functional.test.media.FullTeachingEndToEndRESTTests'
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:628)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:285)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:250)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1256)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1105)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:939)
[ERROR] 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:126)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:328)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:316)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:174)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.access$000(MojoExecutor.java:75)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor$1.run(MojoExecutor.java:162)
[ERROR] 	at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute(DefaultMojosExecutionStrategy.java:39)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:159)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:105)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:73)
[ERROR] 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:53)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:118)
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:261)
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:173)
[ERROR] 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:101)
[ERROR] 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:906)
[ERROR] 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:283)
[ERROR] 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:206)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:283)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:226)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:407)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:348)
[ERROR] 
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

--- a/pom.xml
+++ b/pom.xml
@@ -26,44 +26,24 @@
     </developers>
 
     <properties>
-        <!-- This oversuscribe the ENV variable-->
-        <RETORCH_TESTENV>false</RETORCH_TESTENV>
-
-        <!-- Java 8 -->
-        <java.version>11</java.version>
-        <maven.compiler.source>${java.version}</maven.compiler.source>
-        <maven.compiler.target>${java.version}</maven.compiler.target>
-        <maven.compiler.version>3.13.0</maven.compiler.version>
-        <!-- Encoding -->
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>1.8</maven.compiler.target>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
 
-        <!-- Test dependencies version -->
-        
-        <junit.platform.version>1.0.0</junit.platform.version>
-        
-        <selenium-jupiter.version>5.1.1</selenium-jupiter.version>
-
-        <junit-jupiter.version>5.11.3</junit-jupiter.version>
-        
-        <webdrivermanager.version>5.9.2</webdrivermanager.version>
-        
-        <retorch.version>1.1.0</retorch.version>
-        
-        <maven-surefire-plugin.version>3.5.1</maven-surefire-plugin.version>
-        
-        <jacoco-maven-plugin.version>0.8.7</jacoco-maven-plugin.version>
+        <surefire.version>3.5.2</surefire.version>
 
         <slf4j-api.version>2.0.16</slf4j-api.version>
 
-        <log4j-slf4j2-impl.version>2.24.0</log4j-slf4j2-impl.version>
+        <log4j-slf4j2-impl.version>2.24.1</log4j-slf4j2-impl.version>
 
-        <!-- Plugins Jacoco -->
-        <jacoco.version>0.8.12</jacoco.version>
-        <sonar.java.coveragePlugin>jacoco</sonar.java.coveragePlugin>
-        <sonar.dynamicAnalysis>reuseReports</sonar.dynamicAnalysis>
-        <sonar.jacoco.reportPath>${project.basedir}/../target/jacoco.exec</sonar.jacoco.reportPath>
-        <sonar.language>java</sonar.language>
+        <junit-jupiter-api.version>5.11.3</junit-jupiter-api.version>
+
+        <selenium-java.version>4.26.0</selenium-java.version>
+
+        <webdrivermanager.version>5.9.2</webdrivermanager.version>
+
+        <retorch-annotations.version>1.1.0</retorch-annotations.version>
+
     </properties>
 
     <scm>
@@ -86,43 +66,6 @@
     </distributionManagement>
 
     <dependencies>
-        <!-- Test dependencies -->
-        <dependency>
-            <groupId>org.junit.jupiter</groupId>
-            <artifactId>junit-jupiter-engine</artifactId>
-            <version>${junit-jupiter.version}</version>
-            <scope>test</scope>
-        </dependency>
-        <dependency>
-            <groupId>org.junit.jupiter</groupId>
-            <artifactId>junit-jupiter-params</artifactId>
-            <version>${junit-jupiter.version}</version>
-            <scope>test</scope>
-        </dependency>
-        <!--	As of version 4, Selenium-Jupiter does not include Selenium WebDriver as a transitive dependency.-->
-        <dependency>
-            <groupId>org.seleniumhq.selenium</groupId>
-            <artifactId>selenium-java</artifactId>
-            <version>4.25.0</version>
-        </dependency>
-        <dependency>
-            <groupId>org.junit.jupiter</groupId>
-            <artifactId>junit-jupiter-api</artifactId>
-            <version>${junit-jupiter.version}</version>
-            <scope>test</scope>
-        </dependency>
-        <dependency>
-            <groupId>io.github.bonigarcia</groupId>
-            <artifactId>selenium-jupiter</artifactId>
-            <version>${selenium-jupiter.version}</version>
-            <scope>test</scope>
-        </dependency>
-        <dependency>
-            <groupId>io.github.bonigarcia</groupId>
-            <artifactId>webdrivermanager</artifactId>
-            <version>${webdrivermanager.version}</version>
-            <scope>test</scope>
-        </dependency>
         <dependency>
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-api</artifactId>
@@ -134,175 +77,176 @@
             <version>${log4j-slf4j2-impl.version}</version>
             <scope>test</scope>
         </dependency>
-        <!-- RETORCH DEPENDENCY-->
+        <dependency>
+            <groupId>org.seleniumhq.selenium</groupId>
+            <artifactId>selenium-java</artifactId>
+            <version>${selenium-java.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+            <version>${junit-jupiter-api.version}</version>
+            <scope>test</scope>
+        </dependency>
         <dependency>
             <groupId>io.github.giis-uniovi</groupId>
             <artifactId>retorch-annotations</artifactId>
-            <version>${retorch.version}</version>
+            <version>${retorch-annotations.version}</version>
+        </dependency>
+        <!--Dependencies related to webdrivers (near future remplaced by selema)-->
+        <dependency>
+            <groupId>io.github.bonigarcia</groupId>
+            <artifactId>webdrivermanager</artifactId>
+            <version>${webdrivermanager.version}</version>
         </dependency>
 
-        <!-- END RETORCH DEPENDENCY-->
+        <!-- Dependency to create the parametrized tests (inclides the annotations) -->
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-params</artifactId>
+            <version>${junit-jupiter-api.version}</version>
+            <scope>test</scope>
+        </dependency>
 
     </dependencies>
     <!-- Plugins -->
 
     <build>
-        <!-- To avoid problems with concurrency we change the directories of the reports and outputs -->
-        <testOutputDirectory>${basedir}/target/test-classes/${dirtarget}</testOutputDirectory>
-        <outputDirectory>${basedir}/target/classes/${dirtarget}</outputDirectory>
-        <plugins>
-            <!-- Jacoco Plugin for Jesus -->
-            <plugin>
-                <groupId>org.jacoco</groupId>
-                <artifactId>jacoco-maven-plugin</artifactId>
-                <version>${jacoco.version}</version>
-                <configuration>
-                    <!--suppress MavenModelInspection -->
-                    <skip>${maven.test.skip}</skip>
-                    <destFile>${basedir}/target/coverage-reports/jacoco-unit.exec</destFile>
-                    <dataFile>${basedir}/target/coverage-reports/jacoco-unit.exec</dataFile>
-                    <output>file</output>
-                    <append>true</append>
-                    <excludes>
-                        <exclude>*MethodAccess</exclude>
-                    </excludes>
-                </configuration>
-                <executions>
-                    <execution>
-                        <id>jacoco-initialize</id>
-                        <goals>
-                            <goal>prepare-agent</goal>
-                        </goals>
-                        <phase>test-compile</phase>
-                    </execution>
-                    <execution>
-                        <id>jacoco-site</id>
-                        <phase>verify</phase>
-                        <goals>
-                            <goal>report</goal>
-                        </goals>
-                    </execution>
-                </executions>
-            </plugin>
-
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-surefire-plugin</artifactId>
-                <version>${maven-surefire-plugin.version}</version>
-
-
-                <configuration>
-                    <!-- This parameter is used to end all the threads (And avoid thread overflown)-->
-                    <shutdown>kill</shutdown>
-                    <!-- This configuration should solve the problem with the starting fork
-                     The error was caused by not enough memory for the app-->
-
-                    <systemPropertyVariables>
-                        <RETORCH_TESTENV>false</RETORCH_TESTENV>
-                    </systemPropertyVariables>
-                    <!-- This configuration excludes the test cases for analysis purposes -->
-
-                    <excludes>com.fullteaching.e2e.no_elastest.common.*</excludes>
-                </configuration>
-            </plugin>
-
-            <!--Added-->
-            <plugin>
-
-
-                <groupId>org.springframework.boot</groupId>
-                <artifactId>spring-boot-maven-plugin</artifactId>
-                <version>3.3.5</version>
-                <!-- ONLY ON DEVELOPMENT -->
-                <dependencies>
-                    <dependency>
-                        <groupId>org.springframework</groupId>
-                        <artifactId>springloaded</artifactId>
-                        <version>1.2.8.RELEASE</version>
-                    </dependency>
-                </dependencies>
-                <!-- ONLY ON DEVELOPMENT -->
-
-            </plugin>
-
-            <plugin>
-                <groupId>org.codehaus.mojo</groupId>
-                <artifactId>exec-maven-plugin</artifactId>
-                <version>3.5.0</version>
-                <configuration>
-                    <mainClass>com.fullteaching.backend.Application</mainClass>
-                </configuration>
-            </plugin>
-
-            <plugin>
-                <groupId>com.spotify</groupId>
-                <artifactId>docker-maven-plugin</artifactId>
-                <version>1.2.2</version>
-                <configuration>
-                    <imageName>codeurjc/${project.artifactId}</imageName>
-                    <imageTags>
-                        <imageTag>latest</imageTag>
-                    </imageTags>
-                    <serverId>docker-hub</serverId>
-                    <dockerDirectory>src/com.fullteaching.e2e.no_elastest.main/docker</dockerDirectory>
-                    <resources>
-                        <resource>
-                            <targetPath>/</targetPath>
-                            <directory>${project.build.directory}</directory>
-                            <include>${project.build.finalName}.war</include>
-                        </resource>
-                        <resource>
-                            <targetPath>/</targetPath>
-                            <directory>src/com.fullteaching.e2e.no_elastest.main/docker</directory>
-                            <include>init.sh</include>
-                        </resource>
-                    </resources>
-                </configuration>
-            </plugin>
-
-            <plugin>
-                <!--suppress MavenModelInspection -->
-                <artifactId>maven-war-plugin</artifactId>
-                <!--suppress MavenModelInspection -->
-                <version>3.4.0</version>
-                <configuration>
-
-                    <webResources>
-                        <resource>
-                            <directory>src/com.fullteaching.e2e.no_elastest.main/ebextensions</directory>
-                            <targetPath>.ebextensions</targetPath>
-                            <filtering>true</filtering>
-                        </resource>
-                    </webResources>
-                </configuration>
-            </plugin>
-
-            <!-- Added  For reports!-->
-
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-surefire-report-plugin</artifactId>
-                <version>3.5.1</version>
-                <executions>
-                    <execution>
-                        <phase>test</phase>
-                        <goals>
-                            <goal>report-only</goal>
-                        </goals>
-                    </execution>
-                </executions>
-            </plugin>
-            <!-- Added  to solve problems with the compiler!-->
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-compiler-plugin</artifactId>
-                <version>${maven.compiler.version}</version>
-                <configuration>
-                    <source>${maven.compiler.source}</source>
-                    <target>${maven.compiler.target}</target>
-                </configuration>
-            </plugin>
-        </plugins>
+        <!-- To avoid problems with concurrency we change the directories of the reports, outputs and build -->
+        <directory>${project.basedir}/target/${tjob_name}</directory>
+        <pluginManagement>
+            <plugins>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-surefire-plugin</artifactId>
+                    <version>${surefire.version}</version>
+                    <configuration>
+                        <testFailureIgnore>false</testFailureIgnore>
+                        <!-- Sets the VM argument line used when unit tests are run under JaCoCo -->
+                        <argLine>${surefireArgLine}</argLine>
+                        <redirectTestOutputToFile>true</redirectTestOutputToFile>
+                        <!-- evita fallo con jenkins slave linux y openjdk: https://stackoverflow.com/questions/23260057/the-forked-vm-terminated-without-saying-properly-goodbye-vm-crash-or-system-exi/53070605 -->
+                        <useSystemClassLoader>false</useSystemClassLoader>
+                        <skipTests>${skipTests}</skipTests>
+                    </configuration>
+                </plugin>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-surefire-report-plugin</artifactId>
+                    <version>${surefire.version}</version>
+                    <executions>
+                        <execution>
+                            <id>ut-reports</id>
+                            <phase>test</phase>
+                            <goals>
+                                <goal>report-only</goal>
+                            </goals>
+                        </execution>
+                    </executions>
+                </plugin>
+                <plugin>
+                    <groupId>org.jacoco</groupId>
+                    <artifactId>jacoco-maven-plugin</artifactId>
+                    <version>0.8.12</version>
+                    <executions>
+                        <execution>
+                            <id>pre-unit-test</id>
+                            <phase>process-test-resources</phase>
+                            <goals>
+                                <goal>prepare-agent</goal>
+                            </goals>
+                            <configuration>
+                                <destFile>${project.build.directory}/coverage-reports/jacoco.exec</destFile>
+                                <propertyName>surefireArgLine</propertyName>
+                            </configuration>
+                        </execution>
+                        <execution>
+                            <id>post-unit-test</id>
+                            <phase>test</phase>
+                            <goals>
+                                <goal>report</goal>
+                            </goals>
+                            <configuration>
+                                <dataFile>${project.build.directory}/coverage-reports/jacoco.exec</dataFile>
+                                <outputDirectory>${project.reporting.outputDirectory}/jacoco</outputDirectory>
+                            </configuration>
+                        </execution>
+                    </executions>
+                </plugin>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-antrun-plugin</artifactId>
+                    <version>3.1.0</version>
+                    <executions>
+                        <!-- Aggregate junit style reports -->
+                        <execution>
+                            <id>junit-reports</id>
+                            <phase>test</phase>
+                            <configuration>
+                                <target unless="skipTests">
+                                    <junitreport>
+                                        <fileset dir="${project.basedir}/target/surefire-reports"
+                                                 erroronmissingdir="false">
+                                            <include name="**/*.xml"/>
+                                        </fileset>
+                                        <report format="frames"
+                                                todir="${project.reporting.outputDirectory}/junit-frames"/>
+                                        <report format="noframes"
+                                                todir="${project.reporting.outputDirectory}/junit-noframes"/>
+                                    </junitreport>
+                                </target>
+                            </configuration>
+                            <goals>
+                                <goal>run</goal>
+                            </goals>
+                        </execution>
+                    </executions>
+                    <dependencies>
+                        <dependency>
+                            <groupId>org.apache.ant</groupId>
+                            <artifactId>ant-junit</artifactId>
+                            <version>1.10.15</version>
+                        </dependency>
+                        <dependency>
+                            <groupId>org.apache.ant</groupId>
+                            <artifactId>ant-trax</artifactId>
+                            <version>1.8.0</version>
+                        </dependency>
+                    </dependencies>
+                </plugin>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-source-plugin</artifactId>
+                    <version>3.3.1</version>
+                    <executions>
+                        <execution>
+                            <id>attach-sources</id>
+                            <goals>
+                                <goal>jar-no-fork</goal>
+                            </goals>
+                        </execution>
+                    </executions>
+                </plugin>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-javadoc-plugin</artifactId>
+                    <version>3.11.1</version>
+                    <configuration>
+                        <quiet>true</quiet>
+                        <doclint>none</doclint>
+                    </configuration>
+                    <executions>
+                        <execution>
+                            <id>attach-javadocs</id>
+                            <phase>package</phase>
+                            <goals>
+                                <goal>jar</goal>
+                            </goals>
+                        </execution>
+                    </executions>
+                </plugin>
+            </plugins>
+        </pluginManagement>
     </build>
 
 </project>

--- a/src/test/java/com/fullteaching/e2e/no_elastest/functional/test/UserTest.java
+++ b/src/test/java/com/fullteaching/e2e/no_elastest/functional/test/UserTest.java
@@ -8,9 +8,7 @@ import com.fullteaching.e2e.no_elastest.common.exception.NotLoggedException;
 import com.fullteaching.e2e.no_elastest.utils.ParameterLoader;
 import giis.retorch.annotations.AccessMode;
 import giis.retorch.annotations.Resource;
-import io.github.bonigarcia.seljup.SeleniumJupiter;
 import org.junit.jupiter.api.DisplayName;
-import org.junit.jupiter.api.extension.ExtendWith;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.Arguments;
 import org.junit.jupiter.params.provider.MethodSource;
@@ -20,7 +18,6 @@ import java.util.stream.Stream;
 
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
-@ExtendWith(SeleniumJupiter.class)
 class UserTest extends BaseLoggedTest {
 
     public static Stream<Arguments> data() throws IOException {

--- a/src/test/java/com/fullteaching/e2e/no_elastest/functional/test/teacher/CourseTeacherTest.java
+++ b/src/test/java/com/fullteaching/e2e/no_elastest/functional/test/teacher/CourseTeacherTest.java
@@ -12,8 +12,6 @@ import com.fullteaching.e2e.no_elastest.utils.ParameterLoader;
 import com.fullteaching.e2e.no_elastest.utils.Wait;
 import giis.retorch.annotations.AccessMode;
 import giis.retorch.annotations.Resource;
-import io.github.bonigarcia.seljup.SeleniumJupiter;
-import org.junit.jupiter.api.extension.ExtendWith;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.Arguments;
 import org.junit.jupiter.params.provider.MethodSource;
@@ -30,7 +28,6 @@ import static com.fullteaching.e2e.no_elastest.common.Constants.*;
 import static com.fullteaching.e2e.no_elastest.common.CourseNavigationUtilities.checkIfCourseExists;
 import static org.junit.jupiter.api.Assertions.*;
 
-@ExtendWith(SeleniumJupiter.class)
 class CourseTeacherTest extends BaseLoggedTest {
 
     public static Stream<Arguments> data() throws IOException {

